### PR TITLE
Use system libcrypto

### PR DESCRIPTION
**Pull request by ThadHouse**
Try to see if we can statically link openssl, but dynamically link libcrypto.

### CODE REVIEW

**Comment by nibanks** [comment]
Since this seems to work, let's make this build-time configurable, and leave the default to completely statically link everything for now.

File: scripts/build.ps1 [hunk 1]
```diff
@@ -224,7 +236,12 @@ $BaseBuildDir = Join-Path $RootDir "build"
 $ArtifactsDir = Join-Path $BaseArtifactsDir "bin" $Platform
 $BuildDir = Join-Path $BaseBuildDir $Platform
 
-$ArtifactsDir = Join-Path $ArtifactsDir "$($Arch)_$($Config)_$($Tls)"
+if ("" -eq $ExtraArtifactDir) {
+    $ArtifactsDir = Join-Path $ArtifactsDir "$($Arch)_$($Config)_$($Tls)"
+} else {
+    $ArtifactsDir = Join-Path $ArtifactsDir "$($Arch)_$($Config)_$($Tls)_$($ExtraArtifactDir)"
+}
+
```
Comment (nibanks) [inline scripts/build.ps1:244]: Theoretically, a lot of tools (e.g. test.ps1) would need to be updated as well, right?

Comment (ThadHouse) [reply scripts/build.ps1:244]: If we wanted to run the builds, yes. I can do that either here or in a separate PR.

Comment (nibanks) [reply scripts/build.ps1:244]: We should add a test for the system libcrypto variant of openssl on Linux. Please add that, and only update the scripts that need to be updated for that to happen.

**Review by nibanks — APPROVED** [review]

**Pull request merged**